src/Types.ml

```diff
@@ -58,41 +64,55 @@ let typeof (v : value) : typ =
   match v with
   | VInt(_)     -> TInt
   | VBool(_)    -> TBool
+  | VList(_)    -> TList
   | _           -> TUnknown
 
 let of_int i = (VInt i)
 let of_bool b = (VBool b)
+let of_list l = (VList l)
 
 let vtrue = VBool true
 let vfalse = VBool false
 
 let from_int = function | VInt(i) -> i | _ -> raise (Internal_Exn "")
 let from_bool = function | VBool(b) -> b | _ -> raise (Internal_Exn "")
+let from_list = function | VList(l) -> l | _ -> raise (Internal_Exn "")
 
-let deserialize_value_to ~(t : typ) (s : string) : value option =
+let rec list_of_string s = 
+  let s = String.sub ~pos: 1 ~len: (String.length s - 2) s in
+  String.split ~on:',' s 
+  |> List.filter_map ~f:deserialize_value
```

Comment:
1. Please move this to `Utils` as `List.of_string`
2. Replace `,` with a parameter `sep:';'`

Reply:
It uses deserialize_value which is a funciton in Types.ml though. 
To resolve this I added a second parameter to list_of_string which takes a function to fill the place of deserialize_value.